src/dram_controller.cc

```diff
@@ -317,12 +412,19 @@ void MEMORY_CONTROLLER::end_phase(unsigned cpu)
 
 void DRAM_CHANNEL::end_phase(unsigned /*cpu*/) { roi_stats = sim_stats; }
 
+bool DRAM_ADDRESS_MAPPING::is_collision(champsim::address a, champsim::address b) const
+{
+  //collision if everything but offset matches
+  champsim::data::bits offset_bits = champsim::data::bits{champsim::size(get<SLICER_OFFSET_IDX>(address_slicer))};
+  return(a.slice_upper(offset_bits) == b.slice_upper(offset_bits));
+}
+
 void DRAM_CHANNEL::check_write_collision()
 {
   for (auto wq_it = std::begin(WQ); wq_it != std::end(WQ); ++wq_it) {
     if (wq_it->has_value() && !wq_it->value().forward_checked) {
-      auto checker = [check_val = champsim::block_number{wq_it->value().address}](const auto& pkt) {
-        return pkt.has_value() && champsim::block_number{pkt->address} == check_val;
+      auto checker = [chan = this, check_val = wq_it->value().address](const auto& pkt) {
+        return pkt.has_value() && chan->address_mapping.is_collision(pkt.value().address,check_val);
```

Comment:
```suggestion
      auto checker = [addr_map=address_mapping, check_val = wq_it->value().address](const auto& pkt) {
        return pkt.has_value() && addr_map.is_collision(pkt.value().address,check_val);
```

This would be a more minimal capture